### PR TITLE
Simulate secondary

### LINKED ISSUE #563: stan model implementation of `simulate_secondary`
`simulate_secondary` is a nice function for convolutions with potentially time-varying delays and scaling, implemented in R.

Whilst this is great I think it would be nice to have a `simulate_secondary` function that mirrors `simulate_infections` and uses the corresponding stan model to simulate from given parameters. This could benefit from further model updates (and already e.g. can handle gamma distributed delays) and e.g. the new distribution interface but for the moment wouldn't be able to have time-varying convolution parameters. Implementing this would also enable the deprecations mentioned in #520.

We could
1. replace `simulate_secondary` with the function outlined above
2. keep `simulate_secondary` and add the function outlined above with different name
3. rename `simulate_secondary` and add the function outlined above as `simulate_secondary`
4. not do anything and keep things as they are

My initial inclination would be (3) with an aim to eventually deprecate the function when time-varying convolutions become available in the stan model.

### PR DESCRIPTION
**Pull request by sbfnk**
## Description

This PR closes #563 by introducing a new function `simulate_secondary()`. The existing function of that named has been renamed to `convolve_and_scale()`

## Initial submission checklist

- [X] My PR is based on a package issue and I have explicitly linked it.
- [X] I have tested my changes locally (using `devtools::test()` and `devtools::check()`).
- [X] I have added or updated unit tests where necessary.
- [X] I have updated the documentation if required and rebuilt docs if yes (using `devtools::document()`).
- [X] I have followed the established coding standards (and checked using `lintr::lint_package()`).
- [X] I have added a news item linked to this PR.

## After the initial Pull Request 

- [ ] I have reviewed Checks for this PR and addressed any issues as far as I am able.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -28,6 +28,7 @@
 * Added the possibility of specifying fixed overdispersion. By @sbfnk in #560 and reviewed by @seabbs.
 * The example in `estimate_truncation()` has been simplified. The package now ships with a dataset `example_truncated`, which is used in the `estimate_truncation()` example and tests. The steps for creating the `example_truncated` is stored in `./data-raw/estimate-truncation.R`. By @jamesmbaazam in #584 and reviewed by @seabbs and @sbfnk.
 * Tests have been updated to only set random seeds before snapshot tests involving random number generation, and unset them subsequently. By @sbfnk in #590 and reviewed by @seabbs.
+* A function `simulate_secondary()` was added to simulate from parameters of the `estimate_secondary` model. A function of the same name that was previously based on a reimplementation of that model in R with potentially time-varying scalings and delays has been renamed to `convolve_and_scale()`. By @sbfnk in 591.
```
Comment (seabbs) [inline NEWS.md:31]:
```suggestion
* A function `simulate_secondary()` was added to simulate from parameters of the `estimate_secondary` model. A function of the same name that was previously based on a reimplementation of that model in R with potentially time-varying scalings and delays has been renamed to `convolve_and_scale()`. By @sbfnk in #591 and reviewed by @seabbs.
```

File: R/simulate_secondary.R [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+#'   (column `date`). Column `primary` must be numeric and `date` must be in
+#'   date format.  it will be assumed that `primary` is zero on the missing
+#'   days.
+#' @inheritParams simulate_infections
+#' @inheritParams estimate_secondary
+#' @importFrom checkmate assert_data_frame assert_date assert_numeric
+#'   assert_subset
+#' @return A data.table of simulated secondary observations (column `secondary`)
+#'   by date.
+#' @author Sebastian Funk
```
Comment (seabbs) [inline R/simulate_secondary.R:25]:
```suggestion
```

**Review by seabbs — DISMISSED** [review]
LGTM. A bit of a painful merge I think.

**Comment by sbfnk** [comment]
All changes since last review were merges / minor doc changes so merging as is.

**Pull request merged**